src/main/java/uk/gov/hmcts/reform/pbis/notify/NotificationClientProvider.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.pbis.notify;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.pbis.ServiceNotFoundException;
+import uk.gov.hmcts.reform.pbis.config.ApplicationConfig;
+import uk.gov.hmcts.reform.pbis.model.EmailTemplateMapping;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientApi;
+
+@Component
+public class NotificationClientProvider {
+
+    private final boolean useClientStub;
+
+    private final Map<String, NotificationClient> notificationClientsByService;
+
+    @Autowired
+    public NotificationClientProvider(ApplicationConfig config) {
+        useClientStub = config.getUseNotifyClientStub();
+        notificationClientsByService = getApiClientsByService(config.getEmailTemplateMappings());
+    }
+    
+    public NotificationClientApi getClient(String service) {
+        if (useClientStub) {
+            return new NotificationClientStub();
+        } else if (notificationClientsByService.containsKey(service)) {
+            return notificationClientsByService.get(service);
+        } else {
+            throw new ServiceNotFoundException(
+                String.format("Failed to find Notify API client for service %s", service)
+            );
+        }
+    }
+
+    private Map<String, NotificationClient> getApiClientsByService(
+        List<EmailTemplateMapping> emailTemplateMappings
+    ) {
+        return emailTemplateMappings
+            .stream()
+            .collect(
+                Collectors.toMap(
+                    mapping -> mapping.getService(),
+                    mapping -> new NotificationClient(mapping.getNotifyApiKey())
+                )
+            );
```

Comment:
Perhaps it would be cleaner to initialize clients here based on `useClientStub` flag instead?

Reply:
Make sense, will do

Reply:
done